node/src/state.rs

```diff
@@ -356,6 +368,71 @@ impl State {
         })
     }
 
+    pub fn prevalidate_block(
+        &self,
+        block: &ArcBlockWithHash,
+    ) -> Result<(), BlockPrevalidationError> {
+        let Some((genesis, cur_global_slot)) =
+            None.or_else(|| Some((self.genesis_block()?, self.cur_global_slot()?)))
+        else {
+            // we don't have genesis block. This should be impossible
+            // because we don't even know chain_id before we have genesis
+            // block, so we can't be connected to any peers from which
+            // we would receive a block.
+            return Err(BlockPrevalidationError::GenesisNotReady);
```

Comment:
Maybe we should add bug_condition here?

Reply:
Good pot. Btw do you know what's up with the failed p2p tests? If I run them locally they fail every time too but it is a different one. Changes in this PR shouldn't matter I think